src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -523,6 +523,28 @@ protected HelixWatch3DViewModel(NodeModel model, Watch3DViewModelStartupParams p
 
             SetupScene();
             InitializeHelix();
+            DynamoSelection.Instance.Selection.CollectionChanged += OnSelectionCollectionChanged;
+        }
+
+        private void OnSelectionCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            if (Element3DDictionary == null)
+            {
+                sceneItems = new ObservableElement3DCollection();
+                return;
+            }
+
+            var values = Element3DDictionary.Values.ToList();
+            if (Camera != null)
+            {
+                values.Sort(new Element3DComparer(Camera.Position));
```

Comment:
Why does this need to happen every time the selection changes? Couldn't it be done once when the isolation mode is enabled?

Reply:
Tried that. Doesn't work for some reason.